### PR TITLE
App installation code in node js for teams scope

### PR DESCRIPTION
**Pull request by AbhijeetThakur-msft**
Sample to showcase all scenario for App Installations Lifecycles - Node
It covers the following functionalities:

1. - List apps installed in a team.
2. - Get the specified app installed in a team.
3. - Add (install) an app to a team.
4. - Upgrade the app installed in a team to the latest version.
5. - Remove (uninstall) an app from a team.

### CODE REVIEW

File: samples/graph-app-installation-lifecycle/nodejs/Manifest/manifest.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
+  "$schema": "https://developer.microsoft.com/en-us/json-schemas/teams/v1.8/MicrosoftTeams.schema.json",
+  "manifestVersion": "1.8",
+  "version": "1.0.0",
+  "id": "<<Your Microsoft App Id>>",
```
Comment (v-chetsh) [inline samples/graph-app-installation-lifecycle/nodejs/Manifest/manifest.json:5]: It is not microsoft app id, it is a manifest id.
Please update it accordingly.

File: samples/graph-app-installation-lifecycle/nodejs/Manifest/manifest.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  "packageName": "com.teams.RSC",
```
Comment (v-chetsh) [inline samples/graph-app-installation-lifecycle/nodejs/Manifest/manifest.json:6]: Please check the proper name

File: samples/graph-app-installation-lifecycle/nodejs/Manifest/manifest.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        "team"
+      ]
+    }
+  ],
+  "permissions": [
+    "identity",
+    "messageTeamMembers"
+  ],
+  "validDomains": [
+    "*.ngrok.io"
```
Comment (v-chetsh) [inline samples/graph-app-installation-lifecycle/nodejs/Manifest/manifest.json:40]: Add dummy for deployed app domain

File: samples/graph-app-installation-lifecycle/nodejs/Manifest/manifest.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    "full": "App Installation sample"
+  },
+  "description": {
+    "short": "App Installation",
+    "full": "This example show the App Installation, which includes app creation , update and delete"
+  },
+  "accentColor": "#5558AF",
+  "configurableTabs": [
+    {
+      "configurationUrl": "https://yourngrok.ngrok.io/configure",
```
Comment (v-chetsh) [inline samples/graph-app-installation-lifecycle/nodejs/Manifest/manifest.json:28]: It maybe the deployed app resource URL.
Please update it accordingly.

File: samples/graph-app-installation-lifecycle/nodejs/Readme.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    ngrok http -host-header=rewrite 3978
+    ```
+
+
+1) Register your app with Microsoft identity platform via the Azure AD portal
+    - Your app must be registered in the Azure AD portal to integrate with the Microsoft identity platform and call Microsoft Graph APIs. See [Register an application with the Microsoft identity platform](https://docs.microsoft.com/en-us/graph/auth-register-app-v2).
+    - You need to add following permissions mentioned in the below screenshots to call respective Graph   API
+![](https://user-images.githubusercontent.com/50989436/116188975-e155a300-a745-11eb-9ce5-7f467007e243.png) 
+
+1) Update the `.env` configuration with the ```Microsoft App Id``` and ```App Password```
```
Comment (v-chetsh) [inline samples/graph-app-installation-lifecycle/nodejs/Readme.md:40]: ClientId & Client Secret.
You mentioned in .env file but in doc you are using "Micrisoft App Id" and  "App password"

Please be consistent here.

File: samples/graph-app-installation-lifecycle/nodejs/server.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  res.render('./views/configure');
+});
+
+app.get('/installedAppsList', function (req, res) {
+  var tenantId = req.url.split('=')[1];
+  auth.getAccessToken(tenantId).then(async function (token) {
+    console.log("token from js file : " + token);
+    res.render('./views/installedAppsList', { token: JSON.stringify(token) });
+  });
+
```
Comment (v-chetsh) [inline samples/graph-app-installation-lifecycle/nodejs/server.js:27]: Remove empty line

File: samples/graph-app-installation-lifecycle/nodejs/views/configure.ejs [hunk 7]
```diff
@@ -0,0 +1,46 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>Graph installation app demo node js</title>
+
```
Comment (v-chetsh) [inline samples/graph-app-installation-lifecycle/nodejs/views/configure.ejs:8]: Remove empty line

File: samples/graph-app-installation-lifecycle/nodejs/views/configure.ejs [hunk 8]
```diff
@@ -0,0 +1,46 @@
+<!DOCTYPE html>
+<html lang="en">
+
```
Comment (v-chetsh) [inline samples/graph-app-installation-lifecycle/nodejs/views/configure.ejs:3]: Remove empty line

File: samples/graph-app-installation-lifecycle/nodejs/views/installedAppsList.ejs [hunk 9]
```diff
@@ -0,0 +1,302 @@
...
+                        $("#modalbody").html(html);
+
+                    },
+                    error: function (xhr, textStatus, errorThrown) {
+                        console.log("textStatus: " + textStatus + ", errorThrown:" + errorThrown);
+                    },
+                });
+            });
+        }
+        function upgradeApp(e) {
```
Comment (v-chetsh) [inline samples/graph-app-installation-lifecycle/nodejs/views/installedAppsList.ejs:232]: Add summary for all functions

File: samples/graph-app-installation-lifecycle/nodejs/views/installedAppsList.ejs [hunk 10]
```diff
@@ -0,0 +1,302 @@
...
+                    beforeSend: function (request) {
+                        request.setRequestHeader("Authorization", "Bearer " + token);
+                    },
+                    success: function (profile) {
+                        console.log(profile);
+                        $("#getAppModal").modal('show');
+                        var html = `<span>Name: ${profile.teamsAppDefinition['displayName']} </span> </br> <span>Description: ${profile.teamsAppDefinition['description']} </span>
+                        </br> <span>Version: ${profile.teamsAppDefinition['version']} </span>`;
+                        $("#modalbody").html(html);
+
```
Comment (v-chetsh) [inline samples/graph-app-installation-lifecycle/nodejs/views/installedAppsList.ejs:224]: Remove empty extra line

File: samples/graph-app-installation-lifecycle/nodejs/views/installedAppsList.ejs [hunk 11]
```diff
@@ -0,0 +1,302 @@
...
+                    data: { appId: teamAppInstallationId },
+                    beforeSend: function (request) {
+                        request.setRequestHeader("Authorization", "Bearer " + token);
+                    },
+                    success: function () {
+                        console.log("App deleted succesfully!");
+                        $("#getAppModal").modal('show');
+                        var html = `<span> App deleted successfully</span> </br>`;
+                        $("#modalbody").html(html);
+
```
Comment (v-chetsh) [inline samples/graph-app-installation-lifecycle/nodejs/views/installedAppsList.ejs:287]: Remove empty extra line

File: samples/graph-app-installation-lifecycle/nodejs/views/installedAppsList.ejs [hunk 12]
```diff
@@ -0,0 +1,302 @@
...
+                    },
+                    error: function (xhr, textStatus, errorThrown) {
+                        $("#getAppModal").modal('show');
+                        var html = `<span> Cannot delete this app this app is installed by default.</span> </br>`;
+                        $("#modalbody").html(html);
+                        console.log("textStatus: " + textStatus + ", errorThrown:" + errorThrown);
+                    },
+                });
+            })
+
```
Comment (v-chetsh) [inline samples/graph-app-installation-lifecycle/nodejs/views/installedAppsList.ejs:297]: Remove empty extra line

File: samples/graph-app-installation-lifecycle/nodejs/Readme.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
+# App Installation
```
Comment (v-chetsh) [inline samples/graph-app-installation-lifecycle/nodejs/Readme.md:1]: Check if this project is linked from Root readme files.
 If not, please link it.

**Review by v-chetsh — APPROVED** [review]
All looks good now, approving the PR.

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**